File: python/box/plotlybox.py
import plotly.plotly as py
import plotly.graph_objs as go
import numpy as np
import glob
import pandas as pd
import os
import random

path = "/home/adaboo/Desktop/Masters/sem4/thesis/plot_classification/python/box/data/*.csv"

#ashishbbhayani
#5czLDBg2FdjjsBd1JXLh

#py.sign_in('kumar05', 'IW5WKDoNWeWWon8haT7r')
py.sign_in('ashishbbhayani', '5czLDBg2FdjjsBd1JXLh')
#py.sign_in('zanzy', '6tzTYqsLP9zeG91eamft')

for fname in glob.glob(path):
    print(fname)
    df = (pd.read_csv(fname,encoding= "ISO-8859-1"))
    df1 = df._get_numeric_data()
    graph_name = (os.path.basename(fname))

    final = df1.columns[-1]

    # whole of the csv file
    for a, b in enumerate(df1):

        if (df1.columns.values[a] != final):

            #x_axis = df.iloc[:10, col_1].abs().values.tolist()
            x_axis = df1.iloc[:20, a].values.tolist()

            w_axis = df1.iloc[5:20, a].values.tolist()
            # make them positve
            x_label = df1.columns[a]


            y_axis = df1.iloc[:20, a+ 1].values.tolist()

            z_axis = df1.iloc[5:20, a].values.tolist()
            # make them positve
            y_label = df1.columns[a+1]


            # Create a trace
            trace = go.Box(
                y = x_axis,
                boxpoints=True,
                marker=dict(
                    color='rgb(214, 12, 140)'),
                line=dict(
                    color='rgb(214, 12, 140)')
            )

            # Create a trace
            trace2 = go.Box(
                y=y_axis,
                boxpoints=True,
                marker=dict(
                    color='rgb(9,56,125)'),
                line=dict(
                    color='rgb(9,56,125)')
            )


            # Create a trace
            trace3 = go.Box(
                y=w_axis,
                boxpoints=True,
                marker=dict(
                    color='rgb(0, 128, 128)'),
                line=dict(
                    color='rgb(0, 128, 128)')
            )

            # Create a trace
            trace4 = go.Box(
                y=z_axis,
                boxpoints=True,
                marker=dict(
                    color='rgb(0,0,125)'),
                line=dict(
                    color='rgb(0,0,125)')
            )

            data = [trace,trace2,trace3,trace4]

            layout = go.Layout(width=640, height=480,title='Plot Title',
                xaxis=dict(
                    title='x Axis',
                    titlefont=dict(
                        family='Courier New, monospace',
                        size=16,
                        color='#7f7f7f'
                    )
                ),
                yaxis=dict(
                    title='y Axis',
                    titlefont=dict(
                        family='Courier New, monospace',
                        size=16,
                        color='#7f7f7f'
                    )
                )
            )

            fig = go.Figure(data=data, layout=layout)

            py.image.save_as(fig, filename= 'scatter_plot' + 'plotlyboxesfourhori' + x_label + '.jpeg')

            from IPython.display import Image

            Image('scatter_plot' + '1' + x_label + '.jpeg')

File: R/box/code/plotlybox.R
library(plotly)

Sys.setenv("plotly_username"="kumar05")
Sys.setenv("plotly_api_key"="IW5WKDoNWeWWon8haT7r")
#py.sign_in('kumar05', 'IW5WKDoNWeWWon8haT7r')
#py.sign_in('ashishbbhayani', '5czLDBg2FdjjsBd1JXLh')
#py.sign_in('zanzy', '6tzTYqsLP9zeG91eamft')
#in same folder
#in same folder
setwd("/home/adaboo/Desktop/Masters/sem4/thesis/plot_classification/R/box/data")

out.file<-""

list.files(pattern=".csv$") # use the pattern argument to define a common pattern  for import files with regex. Here: .csv

# create a list from these files
list.filenames<-list.files(pattern=".csv$")

#each file in the folder
for(i in 1:length(list.filenames)){
  
  dataFrame <- read.table(list.filenames[i], header=TRUE, sep=",", stringsAsFactors=FALSE)
  
  #numeric <- dataFrame[,numerics]
  #numeric <- select_if(dataFrame, is_numeric)
  numeric  <- dataFrame[sapply(dataFrame, is.numeric)]
  title <- list.filenames[i]
  title <- sub('\\..*$', '', basename(title))
  print(title)
  
  a=1
  
  b <-c(0,1,2,3)
  
  lin <-c("1","o","b","c","s","S","h")
  
  
  for(n in 1:ncol(numeric)) {
    #each row,column is set
    
    barsize <- sample(b, 1)
    
    
    if(a < ncol(numeric)){
      
      x_label <- names(numeric[a])    
      x_axis <- numeric[1:8,a]
      xn_axis <- numeric[10:35,a]
      xnx_axis <- numeric[15:35,a]
      a <- a+1
      
      y_label <- names(numeric[a])   
      y_axis <- numeric[1:9,a]  
      
      yn_axis <- numeric[8:25,a]
      
      yny_axis <- numeric[9:20,a]
      
      mypath <- file.path("/home/adaboo/Desktop/Masters/sem4/thesis/plot_classification/R/plots")
      
      filename <- paste("otherhorizontalplotly_" , x_label, ".png", sep = "" )
      
      tit = paste("Changed plotybox called ", title, " dataset", sep = "")
      
      f <- list(
        family = "Courier New, monospace",
        size = 18,
        color = "#7f7f7f"
      )
      
      x <- list(
        title = x_label,
        titlefont = f
      )
      y <- list(
        title = y_label,
        titlefont = f
      )
      
      #now <- plotme    
      p <- plot_ly(
        x = x_axis,
        type = "box"
      ) %>%
        
      add_trace(x = y_axis) %>%
        
      add_trace(x = yn_axis) %>%
        
      add_trace(x = xn_axis) %>%
        
      add_trace(x = xnx_axis) %>%
        
      add_trace(x = yny_axis) %>%
        
      layout(title = tit, yaxis = y)
      export(p, file = filename)  
      
    
    }    
    
  }  
}